src/lib/queue.c

```diff
@@ -1 +1,57 @@
 #include "lib/queue.h"
+#include "stdlib/stdalloc.h"
+
+void queue_init(struct queue *q){
+    q->front = NULL;
+    q->back = NULL;
+    q->size=0;
+};
+
+void queue_free(struct queue *q){
+    while (q->front != NULL) {
+        queue_dequeue(q);
+    }
+    free(q);
```

Comment:
The struct itself should not be freed since it's not guaranteed that `q` will be allocated using malloc. It's possible it's been allocated in the data or stack sections, meaning this `free` call will fail.